pkg/exchange/binance/exchange.go

```diff
@@ -183,8 +324,18 @@ func (e *Exchange) QueryAveragePrice(ctx context.Context, symbol string) (float6
 }
 
 func (e *Exchange) NewStream() types.Stream {
+
+	if e.IsMargin {
+		stream := NewStream(e.Client)
+		stream.MarginSettings = e.MarginSettings
+		return stream
+	} else if e.IsFutures {
+		stream := NewFuturesStream(e.futuresClient)
+		stream.FuturesSettings = e.FuturesSettings
+		return stream
+	}
```

Comment:
this can be:

```
stream := NewStream(e.Client, e.futuresClient)
stream.MarginSettings = e.MarginSettings
stream.FuturesSettings = e.FuturesSettings
```

Reply:
WIP